src/core/time_arrays.jl

```diff
@@ -190,41 +191,33 @@ end
 
 # Colon support
 
-function Base.getindex(v::TimestepVector{FixedTimestep{FIRST, STEP}, T}, i::Colon) where {T, FIRST, STEP}
+function Base.getindex(v::TimestepVector{TS, T}, i::Colon) where {TS<:AbstractTimestep, T}
 	return v.data[i]
 end
 
-function Base.getindex(v::TimestepVector{VariableTimestep{TIMES}, T}, i::Colon) where {T, TIMES}
-	return v.data[i]
-end
-
-function Base.setindex!(v::TimestepVector{FixedTimestep{Start, STEP}, T}, val, i::Colon) where {T, Start, STEP}
+function Base.setindex!(v::TimestepVector{TS, T}, val, i::Colon) where {TS<:AbstractTimestep, T}
 	setindex!(v.data, val, i)
 end
-
-function Base.setindex!(v::TimestepVector{VariableTimestep{TIMES}, T}, val, i::Colon) where {T, TIMES}
-	setindex!(v.data, val, i)
-end
-
+	
 # int indexing version supports old-style components and internal functions, not
 # part of the public API
 
-function Base.getindex(v::TimestepVector{FixedTimestep{FIRST, STEP}, T}, i::AnyIndex) where {T, FIRST, STEP}
+function Base.getindex(v::TimestepVector{FixedTimestep{FIRST, STEP}, T}, i::AnyIndex_NonColon) where {T, FIRST, STEP}
 	_throw_int_getindex_depwarning()
 	return v.data[i]
 end
 
-function Base.getindex(v::TimestepVector{VariableTimestep{TIMES}, T}, i::AnyIndex) where {T, TIMES}
+function Base.getindex(v::TimestepVector{VariableTimestep{TIMES}, T}, i::AnyIndex_NonColon) where {T, TIMES}
 	_throw_int_getindex_depwarning()
 	return v.data[i]
 end
 
-function Base.setindex!(v::TimestepVector{FixedTimestep{Start, STEP}, T}, val, i::AnyIndex) where {T, Start, STEP}
+function Base.setindex!(v::TimestepVector{FixedTimestep{Start, STEP}, T}, val, i::AnyIndex_NonColon) where {T, Start, STEP}
 	_throw_int_setindex_depwarning()
 	setindex!(v.data, val, i)
 end
```

Comment:
So why are these functions (and similar ones below for matrices) no longer supporting colon indices? This means that if you try to index with a colon on a TimestepVector or TimestepMatrix, it would just go to the generic TimestepArray methods right? Why do we want that?